### PR TITLE
Correct peer dependencies

### PR DESCRIPTION
**Pull request by zackerydev**
This peer setup is incompatible with shallow installs like `pnpm`

## Bug

- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [x] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

Fixes #44225 #44498 and maybe more I'll find and add later

## Description

Next.js' monorepo setup and dependency listings are incompatible with `pnpm` and specifically `pnpm` mono repos.

https://pnpm.io/symlinked-node-modules-structure

`pnpm` specifically requires all dependencies list in their `package.json` the things they import.

In this instance `@next/font` is missing `next` in its imports array and this causes issues when using `pnpm create next` in a mono repo. You are hit with similar issues to #44225 and #44498 and are unable to run `pnpm run dev`

## Workaround
You can add this to the root package.json as a workaround:

```json5
"pnpm":  {
  "packageExtensions": {
      "@next/font": {
          "next": "13.1.6" // your next version here
        }
   }
}
```

## Future Fix

I have a branch on my fork called `patch-eslint` with a permanent fix. It uses a lint rule to check all instances of this, the lint rule is already enabled on line https://github.com/vercel/next.js/blob/canary/.eslintrc.json#L152

But because of the older version of this lib and lack of `"includeInternal": true` its not auditing cases where the `"next"` import is being resolved from `workspace:*`.

I can include that in my PR if that is in scope.

<details>
<summary>More Potential Issues</summary>
This is the output of the linting rule
```
/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/server.js
  2:16  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  4:17  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  6:24  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  8:14  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/index.ts
    19:1   error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  1835:32  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack-config.ts
  2:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/config/blocks/css/loaders/getCssModuleLocalIdent.ts
  1:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/css-loader/src/utils.ts
  8:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/next-font-loader/index.ts
  6:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/next-image-loader/blur.ts
  1:1  error  'is-animated' should be listed in the project's dependencies. Run 'npm i -S is-animated' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/next-image-loader/index.ts
  1:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/next-metadata-image-loader.ts
  2:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/next-middleware-asset-loader.ts
  1:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/loaders/resolve-url-loader/lib/value-processor.ts
  25:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/plugins/copy-file-plugin.ts
  2:1  error  'loader-utils' should be listed in the project's dependencies. Run 'npm i -S loader-utils' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/build/webpack/plugins/wellknown-errors-plugin/parseNotFoundError.ts
  3:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/app-index.tsx
  6:1  error  'react-server-dom-webpack' should be listed in the project's dependencies. Run 'npm i -S react-server-dom-webpack' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/app-next.ts
  5:3  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  6:3  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/components/router-reducer/fetch-server-response.ts
  2:1  error  'react-server-dom-webpack' should be listed in the project's dependencies. Run 'npm i -S react-server-dom-webpack' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/dev/error-overlay/hot-dev-client.ts
  31:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/image.tsx
  23:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/client/index.tsx
  187:11  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  837:9   error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/lib/patch-incorrect-lockfile.ts
  7:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/lib/try-to-parse-path.ts
  2:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/app-render.tsx
  501:7  error  'react-server-dom-webpack' should be listed in the project's dependencies. Run 'npm i -S react-server-dom-webpack' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/config-shared.ts
  684:27  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/dev/hot-reloader.ts
  4:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/dev/next-dev-server.ts
   59:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  101:7  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/image-optimizer.ts
  7:1  error  'is-animated' should be listed in the project's dependencies. Run 'npm i -S is-animated' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/load-components.ts
  49:35  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  50:18  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  52:24  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/optimize-amp.ts
  7:20  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/render.tsx
  335:7  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/web/sandbox/context.ts
  3:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/server/web/sandbox/sandbox.ts
  2:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/shared/lib/router/utils/path-match.ts
  2:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies
  3:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/shared/lib/router/utils/prepare-destination.ts
  8:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

/Users/zack.griesinger/GitHub/zgriesinger/next.js/packages/next/src/telemetry/events/swc-load-failure.ts
  4:1  error  'next' should be listed in the project's dependencies. Run 'npm i -S next' to add it  import/no-extraneous-dependencies

✖ 44 problems (44 errors, 0 warnings)
```
</details>

### CODE REVIEW

**Review by timneutkens — DISMISSED** [review]

File: packages/font/package.json [hunk 1]
```diff
@@ -20,6 +20,9 @@
     "ncc-fontkit": "ncc build ./fontkit.js -o dist/fontkit",
     "test-pack": "cd ../../ && pnpm test-pack font"
   },
+  "peerDependencies": {
+    "next": "13.1.7-canary.16"
```
Comment (timneutkens) [inline packages/font/package.json:24]:
```suggestion
    "next": "*"
```

**Comment by zgriesinger** [comment]
@timneutkens Any thoughts on the Lint rule? There could be quite a few others missing. This seems to be the most common breakage though.

**Comment by zgriesinger** [comment]
Awesome! Won't add anything than further then, thanks for the review @timneutkens

**Review by cbjjensen — APPROVED** [review]

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 226516c83373faddff572e03e7ba6534df65c1d6

`pnpm testheadless test/development/basic/hmr.test.ts` 
- basic HMR, basePath: "" > Error Recovery > should recover after webpack parse error in an imported file

<details>
<summary>Expand output</summary>

  ● basic HMR, basePath: "" › Error Recovery › should recover after webpack parse error in an imported file

    expect(received).toMatchInlineSnapshot(snapshot)

    Snapshot name: `basic HMR, basePath: "" Error Recovery should recover after webpack parse error in an imported file 2`

    - Snapshot  - 3
    + Received  + 0

    @@ -4,8 +4,5 @@
      | This
      | is
      > }}}
      | invalid
      | js
    -
    - Import trace for requested module:
    - ./components/parse-error.xyz

      756 |             `"Failed to compile"`
      757 |           )
    > 758 |           expect(await getRedboxSource(browser)).toMatchInlineSnapshot(`
          |                                                  ^
      759 |           "./components/parse-error.xyz
      760 |           Module parse failed: Unexpected token (3:0)
      761 |           You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders

      at Object.<anonymous> (development/basic/hmr.test.ts:758:50)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by zgriesinger** [comment]
@timneutkens Are these test failures on me? Anything I need to do further to get this PR through?

**Pull request closed without merging**